src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests/Requests/DataSetQueryCriteriaLocationsValidatorTests.cs

```diff
@@ -0,0 +1,227 @@
+            new DataSetQueryLocationId { Level = "la", Id = "12345" },
+            new DataSetQueryLocationId { Level = "", Id = "12345" },
+        },
+        new DataSetQueryLocation[]
+        {
+            new DataSetQueryLocationId { Level = "NAT", Id = new string('x', 26) },
+            new DataSetQueryLocationCode { Level = "REG", Code = new string('x', 26) },
+            new DataSetQueryLocationProviderUkprn { Ukprn = new string('x', 9) },
+            new DataSetQueryLocationSchoolUrn { Urn = new string('x', 7) },
+        },
+    };
+
+    public class EqTests : DataSetQueryCriteriaLocationsValidatorTests
+    {
+        [Theory]
+        [MemberData(nameof(ValidLocationsSingle))]
+        public void Success(DataSetQueryLocation location)
+        {
+            var query = new DataSetQueryCriteriaLocations { Eq = location };
+
+            _validator.TestValidate(query).ShouldNotHaveAnyValidationErrors();
+        }
+
+        [Theory]
+        [MemberData(nameof(InvalidLocationsSingle))]
+        public void Failure(DataSetQueryLocation location)
+        {
+            var query = new DataSetQueryCriteriaLocations { Eq = location };
+
+            var result = _validator.TestValidate(query);
+
+            Assert.All(result.Errors, error =>
+                Assert.StartsWith("Eq", error.PropertyName));
+        }
+    }
+
+    public class NotEqTests : DataSetQueryCriteriaLocationsValidatorTests
+    {
+        [Theory]
+        [MemberData(nameof(ValidLocationsSingle))]
+        public void Success(DataSetQueryLocation location)
+        {
+            var query = new DataSetQueryCriteriaLocations { NotEq = location };
+
+            _validator.TestValidate(query).ShouldNotHaveAnyValidationErrors();
+        }
+
+        [Theory]
+        [MemberData(nameof(InvalidLocationsSingle))]
+        public void Failure(DataSetQueryLocation location)
+        {
+            var query = new DataSetQueryCriteriaLocations { NotEq = location };
+
+            var result = _validator.TestValidate(query);
+
+            Assert.All(result.Errors, error =>
+                Assert.StartsWith("NotEq", error.PropertyName));
+        }
+    }
+
+    public class InTests : DataSetQueryCriteriaLocationsValidatorTests
+    {
+        [Theory]
+        [MemberData(nameof(ValidLocationsMultiple))]
+        public void Success(params DataSetQueryLocation[] locations)
+        {
+            var testObj = new DataSetQueryCriteriaLocations { In = locations };
+
+            _validator.TestValidate(testObj).ShouldNotHaveAnyValidationErrors();
+        }
+
+        [Theory]
+        [MemberData(nameof(InvalidLocationsMultiple))]
+        public void Failure(DataSetQueryLocation[] locations)
+        {
+            var query = new DataSetQueryCriteriaLocations { In = locations };
+
+            var result = _validator.TestValidate(query);
+
+            Assert.All(result.Errors, error =>
+                Assert.StartsWith("In", error.PropertyName));
```

Comment:
maybe could do with a separate test(s) which covers all the individual error code/messages that are possible. And then, with this test only concerning itself with the error property name, we will have covered all ground.

Reply:
As per my other comment - don't think this is necessary. This would lead to needless redundancy as we're already testing location strings in great depth in `LocationSTringValidatorTests` and in the integration tests.